src/sentry/integrations/slack/tasks.py

```diff
@@ -59,7 +59,16 @@ def _format_value(self, status, rule_id):
 
 @instrumented_task(name="sentry.integrations.slack.search_channel_id", queue="integrations")
 def find_channel_id_for_rule(
```

Comment:
@davidenwang I wish we could just specify the args needed for the updater/creator as part of the `**kwargs` because it's not actually used in the core logic of the function. This function has too many arguments as it is which makes it brittle to future changes. I'll approve this for now so we fix the issue but I think we should refactor this later (with unit tests to catch issues since that was obviously missing) to make it less unwieldy.